kevlar/novel.py

```diff
@@ -96,16 +97,18 @@ def novel(casestream, casecounts, controlcounts, ksize=31, abundscreen=None,
     timer.start()
     nkmers = 0
     nreads = 0
+    nextupdate = updateint
     unique_kmers = set()
-    for n, record in enumerate(casestream, 1):
+    for n, record, mate in kevlar.paired_reader(casestream):
         if skipuntil:
             if record.name == skipuntil:
                 message = 'Found read {:s}'.format(skipuntil)
                 message += ' (skipped {:d} reads)'.format(n)
                 print('[kevlar::novel]', message, file=logstream)
                 skipuntil = False
             continue
-        if n > 0 and n % updateint == 0:
+        if n >= nextupdate:
+            nextupdate += updateint
```

Comment:
It's unlikely that dropping the modulus operation will make a huge difference, but since it's called for every read it could help.